hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1556,33 +1574,14 @@ public List<OzoneFileStatus> listStatus(OmKeyArgs args, boolean recursive,
     return fileStatusList;
   }
 
-  private TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>>
-      getIteratorForKeyInTableCache(
-      boolean recursive, String startKey, String volumeName, String bucketName,
-      TreeMap<String, OzoneFileStatus> cacheKeyMap, String keyArgs,
-      Table<String, OmKeyInfo> keyTable) throws IOException {
-    TableIterator<String, ? extends Table.KeyValue<String, OmKeyInfo>> iterator;
-    Iterator<Map.Entry<CacheKey<String>, CacheValue<OmKeyInfo>>>
-        cacheIter = keyTable.cacheIterator();
-    String startCacheKey = OZONE_URI_DELIMITER + volumeName +
-        OZONE_URI_DELIMITER + bucketName + OZONE_URI_DELIMITER +
-        ((startKey.equals(OZONE_URI_DELIMITER)) ? "" : startKey);
-
-    // First, find key in TableCache
-    listStatusFindKeyInTableCache(cacheIter, keyArgs, startCacheKey,
-        recursive, cacheKeyMap);
-    iterator = keyTable.iterator();
-    return iterator;
-  }
-
   @SuppressWarnings("parameternumber")
-  private void findKeyInDbWithIterator(boolean recursive, String startKey,
-      long numEntries, String volumeName, String bucketName, String keyName,
-      TreeMap<String, OzoneFileStatus> cacheKeyMap, String keyArgs,
-      Table<String, OmKeyInfo> keyTable,
-      TableIterator<String,
-          ? extends Table.KeyValue<String, OmKeyInfo>> iterator)
+  private void listStatusFindKeyInDb(
+      boolean recursive, String startKey, long numEntries, String volumeName,
+      String bucketName, String keyName, TreeMap<String, OzoneFileStatus>
+          cacheKeyMap, String keyArgs, Table<String, OmKeyInfo> keyTable)
```

Comment:
Also this method is quite complex. I wish the entire listStatus implementation can be revisited.